pkg/session_recording/recordings.jsx

```diff
@@ -505,15 +484,17 @@
         }
 
         handleDateSinceChange(date) {
-            this.setState({dateSince: date});
-            this.clearRecordings();
-            this.journalctlRestart();
+            this.setState({dateSince: date}, () => {
+                this.clearRecordings();
+                this.journalctlRestart();
+            });
         }
 
         handleDateUntilChange(date) {
-            this.setState({dateUntil: date});
-            this.clearRecordings();
-            this.journalctlRestart();
+            this.setState({dateUntil: date}, () => {
+                this.clearRecordings();
+                this.journalctlRestart();
+            });
```

Comment:
I agree that passing a callback this way is neat, but I'd also say that this is more difficult to reason about. When exactly this callback will be called? Before or after componentDidUpdate? How many of these callbacks will be called if we change the dates one after another?

Can we perhaps move these to componentDidUpdate? We can check if the dates have changed there, and do the reset and journalctl restart as necessary. BTW, you can also switch to journalctlRestart in componentDidUpdate now.